File: Baekjun/8.1/Parcel/solution.swift
func solution() {
    let inputWN = readLine()!.split(separator: " ").map { Int($0)! }
    let w = inputWN[0]
    let n = inputWN[1]
    let A = readLine()!.split(separator: " ").map { Int($0)! }

    var dp: [Bool] = Array(repeating: false, count: w)
    for i in 0..<n {
        for j in i+1..<n {
            if A[i] + A[j] < w && dp[w - A[i] - A[j]] {
                print("YES")
                return
            }
        }
        
        for j in 0..<i {
            if A[i] + A[j] < w {
                dp[A[i] + A[j]] = true
            }
        }
    }       
    print("NO")
}

solution()

File: Baekjun/8.1/Parcel/solution.py
w, n = map(int, input().split())

A = list(map(int, input().split()))

dp = [False] * w

for i in range(n):
    for j in range(i+1, n):
        if A[i] + A[j] < w and dp[w - A[i] - A[j]]:
            print("YES")
            exit(0)

    for j in range(i):
        if A[i] + A[j] < w:
            dp[A[i] + A[j]] = True

print("NO")
